client-v2/src/components/FrontsEdit/FrontContainer.tsx

```diff
@@ -132,13 +152,58 @@ class Fronts extends React.Component<FrontsComponentProps, ComponentState> {
     this.props.frontsActions.editorCloseFront(this.props.frontId);
   };
 
+  public renameFront = () => {
+    this.setState({
+      frontNameValue: this.getTitle() || '',
+      editingFrontName: true
+    });
+  };
+
+  getTitle = () => {
+    const { selectedFront } = this.props;
+
+    if (selectedFront) {
+      if (selectedFront.metadata && selectedFront.metadata.nameOverride) {
+        return selectedFront.metadata.nameOverride;
+      } else {
+        return startCase(
+          this.props.selectedFront.displayName || this.props.selectedFront.id
+        );
+      }
+    }
+
+    return;
+  };
+
   public render() {
     const { frontId, isFormOpen, isOverviewOpen } = this.props;
-    const title =
-      this.props.selectedFront &&
-      startCase(
-        this.props.selectedFront.displayName || this.props.selectedFront.id
+    const title = this.getTitle();
+
+    const { frontNameValue, editingFrontName } = this.state;
+    const canRename = this.props.selectedFront
+      ? this.props.selectedFront.canRename
+      : false;
+
+    const setName = () => {
```

Comment:
This probably belongs in a method outside of `render`? Or you could write the component with hooks 😁